src/scripts/scripts/zone/caverns_of_time/old_hillsbrad/boss_leutenant_drake.cpp

```diff
@@ -180,24 +186,45 @@ struct boss_lieutenant_drakeAI : public ScriptedAI
         if (Whirlwind_Timer < diff)
         {
             DoCast(me->getVictim(), SPELL_WHIRLWIND);
-            Whirlwind_Timer = 20000+rand()%5000;
-        }else Whirlwind_Timer -= diff;
+            Whirlwind_Timer = urand(18100, 22900);
+        } else Whirlwind_Timer -= diff;
 
         //Fear
         if (Fear_Timer < diff)
         {
             DoScriptText(SAY_SHOUT, me);
             DoCast(me->getVictim(), SPELL_FRIGHTENING_SHOUT);
-            Fear_Timer = 30000+rand()%10000;
-        }else Fear_Timer -= diff;
+            Fear_Timer = urand(25000, 35000);
+        } else Fear_Timer -= diff;
 
         //Mortal Strike
         if (MortalStrike_Timer < diff)
         {
             DoScriptText(SAY_MORTAL, me);
             DoCast(me->getVictim(), SPELL_MORTAL_STRIKE);
-            MortalStrike_Timer = 45000+rand()%5000;
-        }else MortalStrike_Timer -= diff;
+            MortalStrike_Timer = urand(16900, 27800);
+        } else MortalStrike_Timer -= diff;
+        
+        //Exploding Shot
+        if (ExplodingShot_Timer < diff)
+        {
+            if (me->IsNonMeleeSpellCasted(false))
```

Comment:
This can be removed, that code would never be reached because the boss doesn't cast any spells with casting time.